File: 구현/최소직사각형.java
// [프로그래머스] 최소직사각형 (Java)
class Solution {
    static int max = Integer.MIN_VALUE;
    static int subMax = Integer.MIN_VALUE;
    
    static int maxIndex;
    static int minIndex;

    public int solution(int[][] sizes) {
        int answer = 0;
        
        findMax(sizes);
        
        if(maxIndex == 0){
            minIndex = 1;
        }
        
        swapAll(sizes);
        findMaxValue(sizes);
        
        answer = max * subMax;
        return answer;
    }
    
    static void findMax(int[][] sizes){
        for(int i = 0; i < sizes.length; i++){
            for(int j = 0; j < sizes[0].length; j++){
                if(sizes[i][j] > max){
                    max = sizes[i][j];
                    maxIndex = j;
                }
            }
        }
    }
    
    static void swapAll(int[][] sizes){
        for(int i = 0; i < sizes.length; i++){
            if(sizes[i][maxIndex] < sizes[i][minIndex]){
                int tmp = sizes[i][maxIndex];
                sizes[i][maxIndex] = sizes[i][minIndex];
                sizes[i][minIndex] = tmp;
            }
        }
    }
    
    static void findMaxValue(int[][] sizes){
        for(int i = 0; i < sizes.length; i++){
            if(sizes[i][minIndex] > subMax){
                subMax = sizes[i][minIndex];
            }
        }
    }
}

File: 구현/최소직사각형.kt
import kotlin.math.*

// [프로그래머스] 최소직사각형 (Kotlin)
class Solution {
    
    // 최소 가로세로
    var minOne = 0
    var minTwo = 0
    
    fun solution(sizes: Array<IntArray>): Int {
        var answer: Int = 0
        
        minOne = 0
        minTwo = 0
        
        for(i in 0 until sizes.size){
            val one = sizes[i][0]
            val two = sizes[i][1]
            
            if(one > two){
                minOne = max(minOne,one)
                minTwo = max(minTwo,two)
            }else{
                minOne = max(minOne,two)
                minTwo = max(minTwo,one) 
            }
        }
        answer = minOne * minTwo
        
        return answer
    }
}
